src/components/tools/Partials/ChangeLocationStation/utils.js

```diff
@@ -0,0 +1,163 @@
+  mapboxGeocodeParams,
+  caladaptGeocodingEndpoint,
+  formatFeature,
+} from "~/helpers/geocode";
+import { serialize } from "~/helpers/utilities";
+
+export {
+  reverseGeocode as reverseGeocodeAddress,
+  formatFeature,
+} from "~/helpers/geocode";
+
+/**
+ * handleAbortFetch - used to cancel pending fetch requests
+ * @param {AbortController} abortController - AbortController instance
+ * @returns {AbortController}
+ */
+export function handleAbortFetch(abortController) {
+  if (abortController) {
+    abortController.abort();
+  }
+  return new AbortController();
+}
+
+// in most tools these are the default geocode search passed to the MapBox Geocoding API
+const DEFAULT_GEOCODE_SEARCH_TYPES = "address,postcode,neighborhood";
+
+/**
+ * geocodeSearch - gets geographic data for a location search string
+ * @param {string} searchStr – string to geocode
+ * @param {Object} options - additional options
+ * @param {string} options.boundaryType - type of geography, e.g. address, counties, etc.
+ * @param {boolean} options.isStationLayer - does the boundaryType belong to a "stations" group?
+ * @param {AbortSignal} options.signal - signal object from an AbortController instance
+ * @returns {Object} - GeoJSON FeatureCollection
+ */
+export async function geocodeSearch(searchStr, options = {}) {
+  const { boundaryType, signal, isStationLayer } = options;
+  let url;
+  searchStr = sanitizeSearchStr(searchStr, boundaryType);
+  if (boundaryType === "locagrid" || isStationLayer) {
+    url = `${mapboxGeocodingEndpoint}/${searchStr}.json?${serialize({
+      ...mapboxGeocodeParams,
+      autocomplete: true,
+      types: isStationLayer
+        ? DEFAULT_GEOCODE_SEARCH_TYPES.concat(",place,locality")
+        : DEFAULT_GEOCODE_SEARCH_TYPES,
+    })}`;
+  } else {
+    url = `${caladaptGeocodingEndpoint}/${boundaryType}/?${serialize({
+      srs: 4326,
+      search: searchStr,
+      pagesize: 5,
+    })}`;
+  }
+  const response = await fetch(url, {
+    headers: {
+      Accept: "application/json",
+    },
+    ...(signal && { signal }),
+  });
+  const result = await response.json();
+  return result;
+}
+
+/**
+ * reverseGeocodeBoundary - handles querying the Cal-Adapt API for a boundary feature
+ * @param {[number, number]} coordinates
+ * @param {Object} options
+ * @param {string} options.boundaryType - type of geography, e.g. address, counties, etc.
+ * @param {AbortSignal} options.signal - signal object from an AbortController instance
+ * @returns {Object} - GeoJSON FeatureCollection
+ */
+export async function reverseGeocodeBoundary([lng, lat], options = {}) {
+  const { boundaryType, signal } = options;
+  const url = `${caladaptGeocodingEndpoint}/${boundaryType}/?${serialize({
+    srs: 4326,
+    precision: 4,
+    intersects: `Point(${lng} ${lat})`,
+  })}`;
+  const response = await fetch(url, {
+    headers: {
+      Accept: "application/json",
+    },
+    ...(signal && { signal }),
+  });
+  return await response.json();
+}
```

Comment:
Could probably remove this block of code as it's not used in the ChangeLocationStation widget